Better issue templates! (#324)
Atmosphere-NX/Atmosphere: .github/ISSUE_TEMPLATE/feature_request.md

---
name: Feature Request
about: Suggest a new feature for Atmosphère.
#assignees:
---

## Feature Request

> [ If any section does not apply, use nil. ]  

Reviewer: Same thing here. "Use nil" is a strange expression.

[ Lines between [ ] (square brackets) should be removed before posting. ]
### What feature are you suggesting?
#### Overview:
- [ Include the basic, high-level concepts for this feature here. ]

#### Smaller Details: 
- [ These may include specific methods of implementation etc. ]

#### Nature of Request:  
[ Remove all that do not apply to your request. ]
- Addition
  - [ Ex: Addition of certain original features or features from other community projects. ]
  - [ If you are suggesting porting features or including features from other projects, include what license they are distributed under and what, if any libraries those project use. ]
- Change
- Removal

>   - [Ex: Removal of certain features or implementation due to a specific issue/bug or because of low quality code, etc.]

Reviewer: I really don't think we need three separate sections for describing a feature request. Just a "feature description" section would be enough.

Author: The reasoning behind the 3 sections is too weed out generic and low effort requests. I do agree it could be condensed but this section was made based upon my experiences with other repositories and contributions.

> [ Required: Reference the already existing [feature request. ](https://github.com/Atmosphere-NX/Atmosphere/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22features%2Ffeature-request%22) before you make your own request. ]  

Reviewer: We can phrase this more politely.
```suggestion
[ Note: Please [search for previous feature requests](https://github.com/Atmosphere-NX/Atmosphere/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22features%2Ffeature-request%22) to see if your idea has already been suggested before opening a new issue.]  
```

Author: Politeness while ideal, is not the reality. It should be a requirement to reference existing feature requests before a user makes a potentially duplicate issue.

Reviewer: Sure, but I think that my suggested changes convey the same idea more clearly and without coming off as harshly. We (or at least I) want to be somewhat inviting. I think the kind of people who see "Note: Please search for previous feature requests" and think it's less mandatory than "Required: Reference the already existing feature request" aren't the kind of people who would do a good job of searching for that anyway.

Author: Addressed and rephrased.

> [ Failure to do so means your request can and likely will be closed with no warning. ]

Reviewer: Again, I don't think this is actually a reasonable thing for us to do, even if the user didn't bother to search for open feature requests.
```suggestion
```

Author: It should fall upon the user to consult the open feature requests. It is your job as to developer to write code rather than deal with duplicate issues.

Reviewer: It really should, but if users don't check for open feature requests, I believe that it's our tasks as *maintainers* to keep the open issue list clean and provide justification for why we're closing issues. This phrasing is uninviting and implies that we're going to be closing issues without any clarification or justification.

Author: I will definitely rephrase it to something that would inspire confidence in potential contributors. Cheers.

Author: Addressed and rephrased.

[ Lines between [ ] (square brackets) should be removed before posting. ]
### What feature are you suggesting?
#### Overview:
- [ Include the basic, high-level concepts for this feature here. ]

#### Smaller Details: 
- [ These may include specific methods of implementation etc. ]

#### Nature of Request:  
[ Remove all that do not apply to your request. ]
- Addition
  - [ Ex: Addition of certain original features or features from other community projects. ]
  - [ If you are suggesting porting features or including features from other projects, include what license they are distributed under and what, if any libraries those project use. ]
- Change
- Removal
  - [Ex: Removal of certain features or implementation due to a specific issue/bug or because of low quality code, etc.]

[ Required: Reference the already existing [feature request. ](https://github.com/Atmosphere-NX/Atmosphere/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22features%2Ffeature-request%22) before you make your own request. ]  
[ Failure to do so means your request can and likely will be closed with no warning. ]

### What component do you feel this would best fit within?

> - [Fusé](https://github.com/Atmosphere-NX/Atmosphere#components)

Reviewer 2: ``Fusé`` -> ``Fusée``

  - Atmosphère's custom bootloader.
- [Exosphère](https://github.com/Atmosphere-NX/Atmosphere#components)
  - Fully-featured custom secure monitor.
- [Stratosphère](https://github.com/Atmosphere-NX/Atmosphere#components)
    - Custom system modules.
- **[Thermosphère](https://github.com/Atmosphere-NX/Atmosphere#components)** ******
  - Atmosphère's emuNAND implementation.

> - **[Troposphèr. ](https://github.com/Atmosphere-NX/Atmosphere#components)** ******

Reviewer 3: ```suggestion
- **[Troposphère. ](https://github.com/Atmosphere-NX/Atmosphere#components)** ******
```

---
name: Feature Request
about: Suggest a new feature for Atmosphère.
#assignees:
---

## Feature Request

> [ If any section does not apply or you do not have the information needed for that section, use N/A (Not Applicable). ]  

Reviewer: ```suggestion
[ If any section does not apply, replace its contents with "N/A". ]
[ If you do not have the information needed for a section, replace its contents with "Unknown". ]
```

[ Lines between [ ] (square brackets) are to be removed before posting. ]

> [ Please refer to the already existing [feature requests](https://github.com/Atmosphere-NX/Atmosphere/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22features%2Ffeature-request%22) before you make your own request. ]  

Reviewer: ```suggestion
[ Please search for existing [feature requests](https://github.com/Atmosphere-NX/Atmosphere/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22features%2Ffeature-request%22) before you make your own request. ]
```
Remove the `is: open` filter. We don't want duplicates of closed feature requests.
